astropy/coordinates/sites.py

```diff
@@ -0,0 +1,203 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Observatories accessible by the `sites` module originate from the IRAF
+Observatory Database, and are stored in ``data/observatories.json``.
+Longitudes are listed with positive to the West.
+
+Additions or corrections to the observatory list can be submitted via Pull
+Request to the [astropy GitHub repository](https://github.com/astropy/astropy),
+updating the ``observatories.json`` file.
+
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import json
+from difflib import get_close_matches
+
+from ..utils.data import get_pkg_data_contents
+from ..extern import six
+from .earth import EarthLocation
+
+__all__ = ['get_site', 'get_site_names', 'add_site', 'remove_site']
+
+# Observatory database and list of names:
+_site_db = None
+_site_names = []
+
+def _load_sites():
+    """
+    Load observatory database from data/observatories.json and parse them into
+    a dictionary in memory.
+    """
+    global _site_db, _site_names
+    _site_db = dict()
+    db = json.loads(get_pkg_data_contents('data/observatories.json'))
+    for site in db:
+        location = EarthLocation.from_geodetic(db[site]['longitude'],
+                                   db[site]['latitude'],
+                                   db[site]['elevation_meters'])
+        _site_names.append(db[site]['name'])
+        for alias in db[site]['aliases']:
+            _site_db[alias.lower()] = location
+
+def get_site(site_name):
+    """
+    Return an `~astropy.coordinates.EarthLocation` object for known observatory.
+
+    Use `~astropy.coordinates.get_site_names` to see a list of available
+    observatories.
+
+    Parameters
+    ----------
+    site_name : str
+        Name of the observatory.
+
+    Returns
+    -------
+    `~astropy.coordinates.EarthLocation`
+        The location of the observatory.
+
+    See Also
+    --------
+    get_site_names : the list of sites that this function can access
+    add_site : adds a new site to the list
+    remove_site : remove a site by name from the list of sites
+    """
+    if _site_db is None:
+        _load_sites()
```

Comment:
I think a lot of boilerplate can be avoided if `_load_sites()` also actually returns the site DB.  So you can just do `sites = _load_sites()`

Alternatively this could be reorganized into a class that could in principle allow reuse for custom observatory DBs (for example maybe [PANOPTES](http://projectpanoptes.org/v1/)) would like its own observatory DB?  The module level functions could just be aliases to methods on a default observatory DB that exists at the module level (the class could easily handle all the lazy-loading).